File: lifeFull/SpringBootLifeSamadhan/src/main/java/com/lifesamadhan/api/model/Payment.java
package com.lifesamadhan.api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "AssignmentId is required")
    @Min(value = 1, message = "AssignmentId must be a valid value")
    @Column(name = "assignment_id", nullable = false)
    private Long assignmentId;

    @NotNull(message = "Amount is required")
    @DecimalMin(value = "0.0", inclusive = false, message = "Amount must be greater than 0")
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal amount;

    @Pattern(regexp = "^(PENDING|COMPLETED|FAILED|REFUNDED)$", message = "Payment status must be PENDING, COMPLETED, FAILED or REFUNDED")
    @Builder.Default
    @Column(name = "payment_status", nullable = false)
    private String paymentStatus = "PENDING";

    @Column(name = "razorpay_order_id")
    private String razorpayOrderId;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assignment_id", insertable = false, updatable = false)
    @com.fasterxml.jackson.annotation.JsonIgnore
    private ServiceAssignment assignment;
}

File: LifeSamadhan/Backend/Models/Payment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LifeSamadhan.API.Models
{
    [Table("Payments")]
    public class Payment
    {
        [Key]
        [Column("Id")]
        public long Id { get; set; }

        [Column("AssignmentId")]
        public long AssignmentId { get; set; }
        
        [Column("CustomerId")]
        public long CustomerId { get; set; }
        
        [Column("ProviderId")]
        public long ProviderId { get; set; }

        [Column("amount")]
        public double Amount { get; set; }

        [Column("PaymentStatus")]
        public string PaymentStatus { get; set; } = "PENDING";

        [Column("payment_status")]
        public string? PaymentStatusExtra { get; set; } = "PENDING"; 

        [Column("RazorpayOrderId")]
        public string? RazorpayOrderId { get; set; }
        
        [Column("RazorpayPaymentId")]
        public string? RazorpayPaymentId { get; set; }
        
        [Column("RazorpaySignature")]
        public string? RazorpaySignature { get; set; }

        [Column("CreatedAt")]
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        
        [Column("PaidAt")]
        public DateTime? PaidAt { get; set; }
    }
}
